packages/core/src/amazonqFeatureDev/util/files.ts

```diff
@@ -102,29 +104,43 @@ export async function collectFiles(
         const files = respectGitIgnore ? await filterOutGitignoredFiles(rootPath, allFiles) : allFiles
 
         for (const file of files) {
-            try {
-                const fileContent = await SystemUtilities.readFile(file, new TextDecoder('utf8', { fatal: true }))
-                const relativePath = getWorkspaceRelativePath(file.fsPath, { workspaceFolders })
+            const fileStat = await vscode.workspace.fs.stat(file)
```

Comment:
does it make sense to move this right above L113? I don't think we need to check the file stat if we didn't end up reading the file (if it was a binary or whatever)

Reply:
Yep, makes sense! Pushed the change.